src/main/kotlin/com/bardsoftware/papeeria/backend/cosmas/CosmasService.kt

```diff
@@ -0,0 +1,76 @@
+/**
+Copyright 2018 BarD Software s.r.o
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+ */
+package com.bardsoftware.papeeria.backend.cosmas
+
+import com.google.protobuf.ByteString
+import io.grpc.stub.StreamObserver
+
+/**
+ * Special class that can work with requests from client
+ */
+class CosmasService : CosmasGrpc.CosmasImplBase() {
+
+    private val files = mutableMapOf<String, MutableList<ByteString>>()
+
+    override fun getVersion(request: CosmasProto.GetVersionRequest,
+                            responseObserver: StreamObserver<CosmasProto.GetVersionResponse>) {
+        println("Get request for version: ${request.version}")
+        val response = CosmasProto.GetVersionResponse.newBuilder()
+        synchronized(files) {
+            val fileVersions = files[request.fileId]
+            if (fileVersions != null &&
+                    request.version < fileVersions.size &&
+                    request.version >= 0) {
+                response.file = fileVersions[request.version]
+            } else {
+                printErrorInRequest(request)
+            }
+        }
+        responseObserver.onNext(response.build())

```

Comment:
So currently the client gets response with undefined `files` field if file is not found or version number is wrong. It is better to let client clearly know that RPC failed because of client error. There are a few options how to do it: 
* introduce integer error code and string error description in the response object and set it appropriately.
* call [responseObserver.onError](https://grpc.io/grpc-java/javadoc/io/grpc/stub/StreamObserver.html#onError-java.lang.Throwable-) and pass appropriate StatusException 

For this particular place I do not have any strong opinion, so feel free to choose what you like more.

Reply:
Ok, i used second option